### ISSUE TITLE
Vault: mTLS / TLS auth support

### ISSUE THREAD

**Issue by sbaier1** [external-secrets/external-secrets#161]
**Describe the solution you'd like**
I'm planning on using Vault with mTLS [1] + the TLS auth engine [2]. To do so, it must be possible in the SecretStore CRD to configure a private key + certificate that the Vault client should  use when connecting to a Vault server via TLS. 

**What is the added value?**
Use mTLS to securely connect to Vault + authorize against it.

**Give us examples of the outcome**

In addition to the [caBundle](https://github.com/external-secrets/external-secrets/blob/main/deploy/crds/external-secrets.io_secretstores.yaml#L350), it should be possible to specify cert + key files, which are then passed to the Vault client.

**Observations (Constraints, Context, etc):**

None to note, this is a feature request.

[1] https://www.vaultproject.io/docs/configuration/listener/tcp#tls_client_ca_file
[2] https://www.vaultproject.io/docs/auth/cert#tls-certificates-auth-method

**Comment by ricardoptcosta** [comment]
Hi @sbaier1 ,

Thanks for your issue. I finally was able to dedicate some time to it this week. I'm not very familiar with Vault and I'm having trouble getting mTLS and the TLS Authentication Engine going, even before I work on the SecretStore crd. Can you help me setting the keys and certificates in the right place?

This is my understanding of the certificates and keys that we create before we implement the solution.

**Server**  
Private Key - SERVER_PRIVATE_KEY  
Public Key - SERVER_PUBLIC_KEY  
Certificate Signed by CA - SERVER_CERTIFICATE

**Client**  
  Private Key - CLIENT_PRIVATE_KEY  
Public Key - CLIENT_PUBLIC_KEY  
Certificate Signed by CA - CLIENT_CERTIFICATE

**Certificate Authority**  
Private Key - CA_PRIVATE_KEY  
Public Key - CA_PUBLIC_KEY  
Certificate - CA_CERTIFICATE

After having created these files I use them for configuring the Vault. I add to the vault server configuration file the following:

```bash
      listener "tcp" {
        address = "[::]:8200"
        cluster_address = "[::]:8201"
        tls_cert_file = SERVER_CERTIFICATE
        tls_key_file  = SERVER_PRIVATE_KEY
        tls_client_ca_file = CA_CERTIFICATE
      }
```

For authentication, I need to to deploy a certificate for clients to authenticate against:

```bash
vault write auth/cert/certs/web \
    display_name=web \
    policies=web,prod \
    certificate=CA_CERTIFICATE \
    ttl=3600
```

I then login as follows:

```bash
vault login -method=cert \
   -ca-cert=CA_CERTIFICATE \ 
   -client-cert=CLIENT_CERTIFICATE \
   -client-key=CLIENT_PRIVATE_KEY  
```

## Question
I would like to double check whether the keys and certificates are assigned correctly. In particular whether:

```bash
tls_cert_file = SERVER_CERTIFICATE
tls_key_file  = SERVER_PRIVATE_KEY
tls_client_ca_file = CA_CERTIFICATE
```
```bash
certificate=CA_CERTIFICATE
```
```bash
-ca-cert=CA_CERTIFICATE 
-client-cert=CLIENT_CERTIFICATE
-client-key=CLIENT_PRIVATE_KEY 
```

**Comment by moolen** [comment]
This was implemented in #265 :tada:
